### ISSUE TITLE
Filter headers jump sideways and become unusable when there is a horizontal scrollbar

### ISSUE THREAD

**Issue by landroni** [rstudio/DT#599]
When using filters positioned on top of the table, and the table is wide enough for the horizontal scrollbar to appear, the right-most filters -- esp. numeric and factor -- become unusable. 

Here's a minimal example: 

    ---
    title: "MRE"
    author: ""
    output: 
      flexdashboard::flex_dashboard:
        orientation: columns
        vertical_layout: fill
    runtime: shiny
    ---
    
    Page 1 {data-orientation=columns}
    =======================================================================
    
    Column {data-width=650}
    -----------------------------------------------------------------------
    
    ### Tab 1
    
    ```{r}
    require(dplyr)
    DT::renderDataTable({  
        ##BUG: trying to activate the last 2 filters (factor and numeric) will trigger UI artefacts
        DT::datatable(mtcars %>% 
                          mutate(am = as.character(am), 
                                 gear = as.factor(gear)), 
                      
                      filter = 'top',
                      extensions = c('Scroller'),
                      options = list(scrollY=600, 
                                     scrollX=300, 
                                     scroller = TRUE,
                                     deferRender = TRUE,
                                     pageLength = 250), fillContainer = TRUE)
    })  
    
    ```
    
    Column {data-width=650}
    -----------------------------------------------------------------------
    
    ### Tab 2
    
If you activate the first few filters, they behave as expected (e.g. the filter for `disp`). And things seem to work fine if you activate the `character` filter (i.e. for `am`). 
![screenshot_2018-09-29_11-09-35](https://user-images.githubusercontent.com/1318567/46249123-bf061800-c3d8-11e8-845d-b382e52886de.png)

But things get weird when you activate the `factor` filter, for `gear`. As soon as you click in the box, the table will "visually" disappear. It's just an artefact, by the looks of it, as after you select a level and click 'elsewhere', the table is rendered again. Even so, the behavior is a bit awkward. 
![screenshot_2018-09-29_11-09-43](https://user-images.githubusercontent.com/1318567/46249140-48b5e580-c3d9-11e8-930a-108e4c6a6d61.png)

Unfortunately the `numeric` filters simply become unusable. Unactivated the filters display as expected: 
![screenshot_2018-09-29_11-25-00](https://user-images.githubusercontent.com/1318567/46249223-561f9f80-c3da-11e8-98b8-1768e25ff57d.png)

If you activate the filter for e.g. `carb`, then all the headers jump sideways way to the left: 
![screenshot_2018-09-29_11-19-41](https://user-images.githubusercontent.com/1318567/46249231-7d766c80-c3da-11e8-9001-629083ec51f4.png)

At that moment the `slider` is invisible to the user, and the artefact makes the UI confusing. Resizing the window doesn't seem to help. The only thing that does something is hitting `ESC`, which will then display the slider (although, like for `factor` filters, the table itself will become hidden). And unfortunately this is not the kind of workflow you can teach end-users. 
![screenshot_2018-09-29_11-30-22](https://user-images.githubusercontent.com/1318567/46249272-2a50e980-c3db-11e8-9f3b-72e9527a6581.png)

Is there a way to avoid these UI artefacts? 

-----
```
> sessionInfo()
R version 3.4.4 (2018-03-15)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 14.04.5 LTS

Matrix products: default
BLAS: /usr/lib/libblas/libblas.so.3.0
LAPACK: /usr/lib/lapack/liblapack.so.3.0

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C               LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8    LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C             LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] DT_0.4              data.table_1.10.4-2 shiny_1.0.5        

loaded via a namespace (and not attached):
 [1] Rcpp_0.12.18          digest_0.6.15         rprojroot_1.3-2       mime_0.5             
 [5] R6_2.2.2              xtable_1.8-2          jsonlite_1.5          backports_1.1.2      
 [9] magrittr_1.5          evaluate_0.10.1       stringi_1.1.6         rmarkdown_1.10       
[13] flexdashboard_0.5.1.1 tools_3.4.4           stringr_1.2.0         htmlwidgets_1.0      
[17] crosstalk_1.0.0       httpuv_1.3.5          yaml_2.1.16           compiler_3.4.4       
[21] htmltools_0.3.6       knitr_1.19           
```

**Comment by landroni** [comment]
This issue might be related to #461.

**Comment by shrektan** [comment]
I think it seems related to #554

**Comment by landroni** [comment]
Indeed, looks much like the same issue. (Not the same keywords I was looking for originally.)

**Comment by shrektan** [comment]
Closed by #698 already. Thanks @philibe again.
